pkg/skbn/abs.go

```diff
@@ -0,0 +1,184 @@
+package skbn
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/azure-pipeline-go/pipeline"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+)
+
+var err error
+
+func validateAbsPath(pathSplit []string) error {
+	if len(pathSplit) >= 1 {
+		return nil
+	}
+	return fmt.Errorf("illegal path: %s", filepath.Join(pathSplit...))
+}
+
+func initAbsVariables(split []string) (string, string, string) {
+	account := split[0]
+	container := split[1]
+	path := filepath.Join(split[2:]...)
+
+	return account, container, path
+}
+
+func getNewPipeline() (pipeline.Pipeline, error) {
+	accountName, accountKey := os.Getenv("AZURE_STORAGE_ACCOUNT"), os.Getenv("AZURE_STORAGE_ACCESS_KEY")
+
+	if len(accountName) == 0 || len(accountKey) == 0 {
+		log.Fatal("Either the AZURE_STORAGE_ACCOUNT or AZURE_STORAGE_ACCESS_KEY environment variable is not set")
+	}
+
+	credential, err := azblob.NewSharedKeyCredential(accountName, accountKey)
+
+	if err != nil {
+		log.Fatal("Invalid credentials with error: " + err.Error())
+	}
+	pl := azblob.NewPipeline(credential, azblob.PipelineOptions{})
+
+	return pl, err
+}
+
+func getServiceURL(pl pipeline.Pipeline, accountName string) (azblob.ServiceURL, error) {
+	URL, err := url.Parse(
+		fmt.Sprintf("https://%s.blob.core.windows.net/", accountName))
+
+	surl := azblob.NewServiceURL(*URL, pl)
+	return surl, err
+}
+
+func getContainerURL(pl pipeline.Pipeline, accountName string, containerName string) (azblob.ContainerURL, error) {
+	URL, err := url.Parse(
+		fmt.Sprintf("https://%s.blob.core.windows.net/%s", accountName, containerName))
+
+	curl := azblob.NewContainerURL(*URL, pl)
+	return curl, err
+}
+
+func getBlobURL(curl azblob.ContainerURL, blob string) (azblob.BlockBlobURL, error) {
+	return curl.NewBlockBlobURL(blob), err
+}
+
+func createContainer(ctx context.Context, pl pipeline.Pipeline, curl azblob.ContainerURL) (*azblob.ContainerCreateResponse, error) {
+	cr, err := curl.Create(ctx, azblob.Metadata{}, azblob.PublicAccessNone)
```

Comment:
Would you say that if a container does not exist, is it better to:
1. Create it
2. Return an error

There are good reasons for both, I`ve been thinking of it myself, and chose to return an error.
The main reason for that is that most people manage their cloud infrastructure separately (using Terraform for example), so if the container does not exist, this is most likely a typo...

What do you think?

Reply:
Yes, I can change it.

Reply:
Disabled the container creation